src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -510,22 +511,31 @@ private function addServiceOverriddenGetters($id, Definition $definition)
             }
             if (!$r->isFinal()) {
                 if (0 < $r->getNumberOfParameters()) {
-                    $getters = implode('__construct($container'.$this->salt.', ', explode('(', $this->generateSignature($r), 2));
+                    $constructor = implode('($container'.$this->salt.', ', explode('(', InheritanceProxyHelper::getSignature($r, $call), 2));
```

Comment:
btw, this tells me that we may be missing a test for this code path, as Travis is green